presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,254 @@
+import static com.facebook.presto.spi.predicate.Marker.Bound.ABOVE;
+import static com.facebook.presto.spi.predicate.Marker.Bound.BELOW;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.util.Objects.requireNonNull;
+
+public class ElasticsearchQueryBuilder
+{
+    private static final Logger log = Logger.get(ElasticsearchQueryBuilder.class);
+
+    private final Duration scrollTime;
+    private final int scrollSize;
+    private final Client client;
+    private final int shard;
+    private final TupleDomain<ColumnHandle> tupleDomain;
+    private final List<ElasticsearchColumnHandle> columns;
+    private final String index;
+    private final String type;
+
+    public ElasticsearchQueryBuilder(List<ElasticsearchColumnHandle> columnHandles, ElasticsearchConnectorConfig config, ElasticsearchSplit split)
+    {
+        columns = requireNonNull(columnHandles, "ElasticsearchColumnHandle is null");
+        requireNonNull(config, "ElasticsearchConnectorConfig is null");
+        requireNonNull(split, "ElasticsearchSplit is null");
+        ElasticsearchTableDescription table = split.getTable();
+        tupleDomain = split.getTupleDomain();
+        index = table.getIndex();
+        type = table.getType();
+        shard = split.getShard();
+        try {
+            Settings settings = Settings.builder().put("cluster.name", table.getClusterName()).build();
+            client = new PreBuiltTransportClient(settings).addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(split.getSearchNode()), table.getPort()));
+        }
+        catch (IOException e) {
+            throw new PrestoException(ELASTIC_SEARCH_CLIENT_ERROR, "Error connecting to Elasticsearch SearchNode: " + split.getSearchNode() + " with port: " + table.getPort(), e);
+        }
+        scrollTime = config.getScrollTime();
+        scrollSize = config.getScrollSize();
+    }
+
+    public SearchRequestBuilder buildScrollSearchRequest()
+    {
+        StringBuilder shardBuilder = new StringBuilder("_shards:");
+        shardBuilder.append(shard);
+        String indices = index != null && !index.isEmpty() ? index : "_all";
+        SearchRequestBuilder searchRequestBuilder = client.prepareSearch(indices)
+                .setTypes(type)
+                .setSearchType(SearchType.QUERY_AND_FETCH)
+                .setScroll(new TimeValue(scrollTime.toMillis()))
+                .setQuery(buildSearchQuery())
+                .setPreference(shardBuilder.toString())
```

Comment:
Using a `StringBuilder` looks like an overkill here as there is a single concat, we can just do `.setPreference("_shards:" + shard)`.